### PR TITLE
Improve test coverage for OpenTelemetry tracing library

### LINKED ISSUE #7396: Improve test coverage for OpenTelemetry tracing package
- Also, add tracing to service mapping.

### PR DESCRIPTION
**Pull request by samvaity**
Closes #7396 

image

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/pom.xml [hunk 1]
```diff
@@ -50,6 +50,18 @@
       <version>0.2.0</version> <!-- {x-version-update;io.opentelemetry:opentelemetry-sdk;external_dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-test</artifactId>
+      <version>1.1.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
```
Comment (alzimmermsft) [inline sdk/core/azure-core-tracing-opentelemetry/pom.xml]: Given that tracing should be released in the same group as Azure Core should this take a dependency on the current version of Azure Core Test?

File: sdk/core/azure-core-tracing-opentelemetry/pom.xml [hunk 2]
```diff
@@ -50,6 +50,18 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-netty</artifactId>
+      <version>1.2.0</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
```
Comment (alzimmermsft) [inline sdk/core/azure-core-tracing-opentelemetry/pom.xml]: Same comment as Azure Core Test

File: sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicyTests.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import com.azure.core.annotation.Get;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.test.implementation.entities.HttpBinJSON;
```
Comment (alzimmermsft) [inline sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicyTests.java]: Is this class worth taking a test dependency on Azure Core Test?

Comment (alzimmermsft) [reply sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicyTests.java]: Especially given it is only used as a response type for a testing interface.

File: sdk/core/azure-core-tracing-opentelemetry/pom.xml [hunk 4]
```diff
@@ -50,6 +50,18 @@
...
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-test</artifactId>
+      <version>1.1.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-netty</artifactId>
```
Comment (alzimmermsft) [inline sdk/core/azure-core-tracing-opentelemetry/pom.xml]: Is anything actually using this dependency?

Comment (samvaity) [reply sdk/core/azure-core-tracing-opentelemetry/pom.xml]: Used for creating a default client [here](https://github.com/Azure/azure-sdk-for-java/pull/7488/files#diff-9c6faa2122f174f015c3e0c7fe8e623bR86).

File: sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicyTests.java (hunk 3 above)
Comment (samvaity) [reply sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicyTests.java]: Removed Azure Core test dependency.

**Pull request merged**